### PR TITLE
Add GetBlock to the Community Providers

### PR DESCRIPTION
**Pull request by JoGetBlock**
## Community Providers

GetBlock (blockchain RPC provider) - Instant API Access to Avalanche - https://getblock.io/nodes/avax/

### CODE REVIEW

**Comment by cgcardona** [comment]
1. Please move your changes below the warning section.
2. In your docs I only see endpoints for the C-Chain. Do you also offer endpoints for the X and P Chains?
3. Create two new sections for mainnet and testnet and list all of your endpoints (both https and wss).

**Comment by JoGetBlock** [comment]
---
description: There is a public API server that allows developers to access the Avalanche platform without having to run a node themselves.
---

# Public API Server

There is a public API server that allows developers to access the Avalanche network without having to run a node themselves. The public API server is actually several [AvalancheGo](https://github.com/ava-labs/avalanchego) nodes behind a load balancer to ensure high availability and high request throughput.

## Using the Public API nodes

The public API server is at `api.avax.network` for Avalanche Mainnet and `api.avax-test.network` for Avalanche Fuji Testnet. To access a particular API, just append the relevant API endpoint, as documented [here](./apis/issuing-api-calls.md). Namely, use the following end points for each chain respectively:

### HTTP

- For C-Chain API, the URL is `https://api.avax.network/ext/bc/C/rpc`.
- For X-Chain API, the URL is `https://api.avax.network/ext/bc/X`.
- For P-Chain API, the URL is `https://api.avax.network/ext/bc/P`.

Note: on Fuji Testnet, use `https://api.avax-test.network/` instead of `https://api.avax.network/`.

### WebSocket

- For C-Chain API, the URL is `wss://api.avax.network/ext/bc/C/ws`.

Note: on Fuji Testnet, the URL is `wss://api.avax-test.network/ext/bc/C/ws`.

## Supported APIs

The public API server supports all the API endpoints that make sense to be available on a public-facing service, including APIs for the [X-Chain](./apis/x-chain.md), [P-Chain](./apis/p-chain.md), [C-Chain](./apis/c-chain.md), and full archival for the Primary Network. However, it does not support [Index APIs](./apis/index-api.md), which includes the X-Chain API's `getAddressTxs` method.

For a full list of available APIs see [here](./apis/README.md).

## Limitations

The public API only supports C-Chain websocket API calls for API methods that don't exist on the C-Chain's HTTP API.

For batched C-Chain requests on the public api node, the maximum number of items is 40. We are working on to support a larger batch size.

The maximum number of blocks to serve per `getLogs` request is 2048, which is set by [`api-max-blocks-per-request`](../../nodes/maintain/chain-config-flags.md#api-max-blocks-per-request-int).

## Sticky sessions

Requests to the public API server API are distributed by a load balancer to an individual node. As a result, consecutive requests may go to different nodes. That can cause issues for some use cases. For example, one node may think a given transaction is accepted, while for another node the transaction is still processing. To work around this, you can use 'sticky sessions', as documented [here](https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials). This allows consecutive API calls to be routed to the same node.

If you're using [AvalancheJS](../avalanchejs/README.md) to access the public API, simply set the following in your code:

```javascript
avalanche.setRequestConfig("withCredentials", true)
```

## Availability

Usage of public API nodes is free and available to everyone without any authentication or authorization. Rate limiting is present, but many of the API calls are cached, and the rate limits are quite high. If your application is running up against the limits, please [contact us](https://chat.avalabs.org).

## Support

If you have questions, problems or suggestions, come [talk to us](https://chat.avalabs.org/).

## Community Providers

:::warning Disclaimer

Ava Labs offers no warranty/guarantee on the quality or uptime of these 3rd party RPC providers
:::

GetBlock (blockchain RPC provider) - Instant API Access to Avalanche 

Testnet endpoints:

wss://avax.getblock.io/api_key/testnet/ext/bc/C/ws?api_key=
https://avax.getblock.io/api_key/testnet/ext/bc/C/ws?api_key=

Mainnet Endpoint:

wss://avax.getblock.io/api_key/mainnet/ext/bc/C/ws?api_key=
https://avax.getblock.io/api_key/mainnet/ext/bc/C/ws?api_key=

File: docs/apis/avalanchego/public-api-server.md [hunk 1]
```diff
@@ -58,7 +58,9 @@ If you have questions, problems or suggestions, come [talk to us](https://chat.a
 
 ## Community Providers
 
+GetBlock (blockchain RPC provider) - Instant API Access to Avalanche - https://getblock.io/nodes/avax/
```
[the remark this answers is not in the document]
Comment (JoGetBlock) [reply docs/apis/avalanchego/public-api-server.md:61]: 1. Got it
2. Yes, we only work with C-chain now
3. Thank you

[the remark this answers is not in the document]
Comment (yulin-dong) [reply docs/apis/avalanchego/public-api-server.md:61]: Please make the changes to doc, not just comment on this PR.

**Comment by yulin-dong** [comment]
> ## description: There is a public API server that allows developers to access the Avalanche platform without having to run a node themselves.
> # Public API Server
> There is a public API server that allows developers to access the Avalanche network without having to run a node themselves. The public API server is actually several [AvalancheGo](https://github.com/ava-labs/avalanchego) nodes behind a load balancer to ensure high availability and high request throughput.
> 
> ## Using the Public API nodes
> The public API server is at `api.avax.network` for Avalanche Mainnet and `api.avax-test.network` for Avalanche Fuji Testnet. To access a particular API, just append the relevant API endpoint, as documented [here](./apis/issuing-api-calls.md). Namely, use the following end points for each chain respectively:
> 
> ### HTTP
> * For C-Chain API, the URL is `https://api.avax.network/ext/bc/C/rpc`.
> * For X-Chain API, the URL is `https://api.avax.network/ext/bc/X`.
> * For P-Chain API, the URL is `https://api.avax.network/ext/bc/P`.
> 
> Note: on Fuji Testnet, use `https://api.avax-test.network/` instead of `https://api.avax.network/`.
> 
> ### WebSocket
> * For C-Chain API, the URL is `wss://api.avax.network/ext/bc/C/ws`.
> 
> Note: on Fuji Testnet, the URL is `wss://api.avax-test.network/ext/bc/C/ws`.
> 
> ## Supported APIs
> The public API server supports all the API endpoints that make sense to be available on a public-facing service, including APIs for the [X-Chain](./apis/x-chain.md), [P-Chain](./apis/p-chain.md), [C-Chain](./apis/c-chain.md), and full archival for the Primary Network. However, it does not support [Index APIs](./apis/index-api.md), which includes the X-Chain API's `getAddressTxs` method.
> 
> For a full list of available APIs see [here](./apis/README.md).
> 
> ## Limitations
> The public API only supports C-Chain websocket API calls for API methods that don't exist on the C-Chain's HTTP API.
> 
> For batched C-Chain requests on the public api node, the maximum number of items is 40. We are working on to support a larger batch size.
> 
> The maximum number of blocks to serve per `getLogs` request is 2048, which is set by [`api-max-blocks-per-request`](../../nodes/maintain/chain-config-flags.md#api-max-blocks-per-request-int).
> 
> ## Sticky sessions
> Requests to the public API server API are distributed by a load balancer to an individual node. As a result, consecutive requests may go to different nodes. That can cause issues for some use cases. For example, one node may think a given transaction is accepted, while for another node the transaction is still processing. To work around this, you can use 'sticky sessions', as documented [here](https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials). This allows consecutive API calls to be routed to the same node.
> 
> If you're using [AvalancheJS](../avalanchejs/README.md) to access the public API, simply set the following in your code:
> 
> ```js
> avalanche.setRequestConfig("withCredentials", true)
> ```
> 
> ## Availability
> Usage of public API nodes is free and available to everyone without any authentication or authorization. Rate limiting is present, but many of the API calls are cached, and the rate limits are quite high. If your application is running up against the limits, please [contact us](https://chat.avalabs.org).
> 
> ## Support
> If you have questions, problems or suggestions, come [talk to us](https://chat.avalabs.org/).
> 
> ## Community Providers
> :::warning Disclaimer
> 
> Ava Labs offers no warranty/guarantee on the quality or uptime of these 3rd party RPC providers :::
> 
> GetBlock (blockchain RPC provider) - Instant API Access to Avalanche
> 
> Testnet endpoints:
> 
> wss://avax.getblock.io/api_key/testnet/ext/bc/C/ws?api_key= https://avax.getblock.io/api_key/testnet/ext/bc/C/ws?api_key=
> 
> Mainnet Endpoints:
> 
> wss://avax.getblock.io/api_key/mainnet/ext/bc/C/ws?api_key= https://avax.getblock.io/api_key/mainnet/ext/bc/C/ws?api_key=

Not sure why copy+paste'd this whole doc here.

**Review by yulin-dong — CHANGES_REQUESTED** [review]
Please fix merge conflict.

**Review by JoGetBlock — COMMENTED** [review]
please provide details about the merge conflict, so I can clarify them with our tech team

**Review by JoGetBlock — COMMENTED** [review]
please provide details about the merge conflict, so I can clarify them with our tech team

**Comment by yulin-dong** [comment]
> please provide details about the merge conflict, so I can clarify them with our tech team

You can see it in the bottom of the PR 
Screen Shot 2022-12-20 at 9 27 22 AM

**Comment by cgcardona** [comment]
Closing in favor of https://github.com/ava-labs/avalanche-docs/pull/1098

**Review by JoGetBlock — COMMENTED** [review]
Ava Labs offers no warranty/guarantee on the quality or uptime of these 3rd party RPC providers
:::

[GetBlock](https://getblock.io/nodes/avax/) (blockchain RPC provider) - Instant API Access to Avalanche

Testnet endpoints:

wss://avax.getblock.io/api_key/testnet/ext/bc/C/ws?api_key=
https://avax.getblock.io/api_key/testnet/ext/bc/C/ws?api_key=

Mainnet Endpoints:

wss://avax.getblock.io/api_key/mainnet/ext/bc/C/ws?api_key=
https://avax.getblock.io/api_key/mainnet/ext/bc/C/ws?api_key=

**Comment by yulin-dong** [comment]
@JoGetBlock if you need to update, please open a new PR instead of commenting on this closed PR.

**Pull request closed without merging**